lex/lexer.go

```diff
@@ -170,7 +202,7 @@ func (l *Lexer) NextToken() Token {
 // Push a named StateFn onto stack.
 func (l *Lexer) Push(name string, state StateFn) {
 	debugf("push %d %v", len(l.stack)+1, name)
-	if len(l.stack) < maxDepth {
+	if len(l.stack) < l.depthLimit || l.depthLimit <= 0 {
```

Comment:
seems like this disables the limit by default now? do we need to initialize `l.depthLimit` to `maxDepth` by default somewhere?

Reply:
yep here https://github.com/lytics/qlbridge/pull/18/files#diff-a5c239c5be2018e8fea27385ffcc6457d8f7901916fca61f0e4df5b80504c592R107

Reply:
i will be more specific rather than <=0. may be just -1 should be good

Reply:
changed it to -1, to prevent any accidental 0 value assignment to depth limit

Reply:
ah nice thanks, missed it